docs/sources/clients/promtail/logrotation.md

```diff
@@ -0,0 +1,132 @@
+
+We recommend (2) as that is the one which works well with Promtail (or any agent) without any dataloss. Now let's understand how do we exactly configure log rotation in different platforms.
+
+## Configure log rotation.
+
+Your logs can be rotated by different components depending on where you are running your application or services. If you are running on Linux bare metal or VM, high chance you will be using [`logrotate`](https://man7.org/linux/man-pages/man8/logrotate.8.html) utility. However if you are running in kubernetes, it's not that obvious who rotates the logs and interestingly it may dependes on what container runtime your kubernetes cluster is using.
+
+### Non kubernetes
+
+Usually in Linux bare metal or VM log rotation is handled by [`logrotate`](https://man7.org/linux/man-pages/man8/logrotate.8.html) utility.
+
+The two different types of log rotation as explained before are supported in `logrotate` as well.
+
+#### Copy and Truncate
+```
+/var/log/apache2/*.log {
+        weekly
+        maxsize 1G
+        copytruncate
+}
+```
+
+Here `copytruncate` mode works exactly like (1) explained above.
+
+#### Rename and Create
+```
+/var/log/apache2/*.log {
+        weekly
+        maxsize 1G
+        create
+}
+```
+Here `create` mode works like (2) explained above. `create` mode is optional because it's the default mode in `logroate`.
+
+`logroate` configs are usually located in `/etc/logrotate/`
+
+It has wide range [different options](https://man7.org/linux/man-pages/man8/logrotate.8.html) for compression, mailing, scripts to run pre and post rotation, etc.
+
+**Recommended: Rename and Create**
+
+### Kubernetes
+
+We can [configure](https://kubernetes.io/docs/concepts/cluster-administration/logging/#log-rotation) `kubelet` running on each node to manage log rotation via two configs
+
+1. `containerLogMaxSize` - It is a quantity defining the maximum size of the container log file before it is rotated. For example: "5Mi" or "256Ki". Default: "10Mi"
+2. `containerLogMaxFiles` - It specifies the maximum number of container log files that can be present for a container. Default: 5
+
+Both should be part of `kubelet` config. Check cloud provider docs for tweaking this configs in managed version of Kubernetes cluster. Examples [GKE](https://cloud.google.com/kubernetes-engine/docs/how-to/node-system-config#create), [AKS](https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration#use-custom-node-configuration) and [EKS](https://eksctl.io/usage/customizing-the-kubelet/#customizing-kubelet-configuration)
+
+NOTE: Log rotation managed by `kubelet` supports only rename + create (for a good reason) and doesn't support copy + truncate.
+
+If `kubelet` is not configured to manage the log rotation, then it's upto CRI (Container Runtime Interface) the cluster is using or can be managed by `logrotate` utility in the kubernetes node itself.
+
+Check your container runtime on your nodes via
+```bash
+$ kubectl get nodes -o wide
```

Comment:
Yea thought about that, but didn't do it because, we cannot guarantee the columns order or may add additional columns in future thus breaking our awk magic.

I could have done it something like 
```
k get nodes -o custom-columns=RUNTIME:.status.nodeInfo.containerRuntimeVersion
```
But again, it also feels like complicated. `kubectl get nodes -o wide` seems simple enough to figure out the container runtime.